management/src/test/java/org/kaazing/gateway/management/jmx/JmxSessionIT.java

```diff
@@ -0,0 +1,134 @@
+                            .done()
+                        .done()
+                        .service()
+                            .property("connector.server.address", "jmx://localhost:2020/")
+                            .type("management.jmx")
+                            .authorization()
+                                .requireRole(ADMIN)
+                            .done()
+                            .realmName("jmxrealm")
+                        .done()
+                        .security()
+                            .keyStore(keyStore)
+                            .keyStorePassword(password)
+                            .keyStoreFile(getKeystoreFileLocation())
+                            .realm()
+                                .name("jmxrealm")
+                                .description("realm for jmx")
+                                .httpChallengeScheme("Application Basic")
+                                .loginModule()
+                                    .type("class:org.kaazing.gateway.management.test.util.TestLoginModule")
+                                    .success("requisite")
+                                .done()
+                            .done()
+                        .done()
+                    .done();
+            // @formatter:on
+            init(configuration);
+        }
+    };
+
+    private JmxRule jmxConnection = new JmxRule(JMX_URI);
+    private TestRule timeout = ITUtil.timeoutRule(10, SECONDS);
+    private TestRule trace = new MethodExecutionTrace();
+
+    @Rule
+    public TestRule chain = RuleChain.outerRule(trace).around(gateway).around(k3po).around(jmxConnection).around(timeout);
+
+    @Specification("client.send.empty.close.frame/handshake.request.and.frame")
+    @ScriptProperty("location 'http://localhost:8001/echo'")
+    @Test
+    public void getActiveSessionCountFromJmx() throws Exception {
+        k3po.finish();
+
+        // to make sure jmx is updated
+        Thread.sleep(1000);
+
+        MBeanServerConnection mbeanServerConn = jmxConnection.getConnection();
+        ObjectName summaryBeansObjectNamePattern = new ObjectName(
+                "org.kaazing.gateway.server.management:root=gateways,subtype=services,serviceType=echo,serviceId=\""
+                        + ECHO_WSN_SERVICE + "\",name=summary,*");
+        Set<ObjectName> mbeanNames = mbeanServerConn.queryNames(summaryBeansObjectNamePattern, null);
+        assertEquals(1, mbeanNames.size());
+        ObjectName summaryBean = mbeanNames.iterator().next();
+        assertEquals(Long.valueOf(1), (Long) mbeanServerConn.getAttribute(summaryBean, "NumberOfCumulativeSessions"));
+        assertEquals(Long.valueOf(0), (Long) mbeanServerConn.getAttribute(summaryBean, "NumberOfCurrentSessions"));
+
+        mbeanNames = mbeanServerConn.queryNames(
+                ObjectName.getInstance("*:serviceType=echo,name=sessions,*"), null);
+        for (ObjectName name : mbeanNames) {
```

Comment:
Why use a for statement? The test will stop running after the first failure. Maybe check that the set size is zero?